packages/typespec-python/test/generic_mock_api_tests/test_typetest_basic.py

```diff
@@ -0,0 +1,29 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import decimal
+
+import pytest
+from typetest.basic import BasicClient
+
+
+@pytest.fixture
+def client():
+    with BasicClient() as client:
+        yield client
+
+
+@pytest.mark.parametrize(
+    "og_name,val",
+    [
+        ("decimal_type", decimal.Decimal("0.33333")),
+        ("decimal128_type", decimal.Decimal("0.33333")),
+    ],
+)
+def test_basic(client: BasicClient, og_name: str, val: dict):
```

Comment:
We could consider test decimal advised by https://github.com/Azure/cadl-ranch/pull/451#discussion_r1396717330

Reply:
Added.